src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const diedInThisCentury = (century)
+    ? people.filter(person => Math.ceil(person.died / 100) === century)
+    : people;
+
+  const onlyMens = diedInThisCentury.filter(person => person.sex === 'm');
+
+  return onlyMens
+    .reduce((sum, person) => person.died - person.born + sum, 0)
```

Comment:
you can make universal function for counting average age for everyone

Reply:
agree let's do this